hack/generator/pkg/codegen/pipeline_augment_status.go

```diff
@@ -0,0 +1,262 @@
+
+type swaggerTypes struct {
+	resources  astmodel.Types
+	otherTypes astmodel.Types
+}
+
+// TODO: is there, perhaps, a way to detect these without hardcoding these paths?
+var skipDirectories = []string{
+	"/examples/",
+	"/quickstart-templates/",
+	"/control-plane/",
+	"/data-plane/",
+}
+
+func loadSwaggerData(ctx context.Context, idFactory astmodel.IdentifierFactory, config *config.Configuration) (swaggerTypes, error) {
+
+	result := swaggerTypes{
+		resources:  make(astmodel.Types),
+		otherTypes: make(astmodel.Types),
+	}
+
+	schemas, err := loadAllSchemas(ctx, config.Status.SchemaRoot)
+	if err != nil {
+		return swaggerTypes{}, err
+	}
+
+	cache := jsonast.MakeOpenAPISchemaCache(schemas)
+
+	for schemaPath, schema := range schemas {
+		// these have already been tested in the loadAllSchemas function so are guaranteed to match
+		outputGroup := swaggerGroupRegex.FindString(schemaPath)
+		outputVersion := swaggerVersionRegex.FindString(schemaPath)
+
+		// see if there is a config override for this file
+		for _, schemaOverride := range config.Status.Overrides {
+			configSchemaPath := path.Join(config.Status.SchemaRoot, schemaOverride.BasePath)
+			if strings.HasPrefix(schemaPath, configSchemaPath) {
+				// found an override: apply it
+				if schemaOverride.Suffix != "" {
+					outputGroup += "." + schemaOverride.Suffix
+				}
+
+				break
+			}
+		}
+
+		extractor := typeExtractor{
+			outputVersion: outputVersion,
+			outputGroup:   outputGroup,
+			idFactory:     idFactory,
+			cache:         cache,
+			config:        config,
+		}
+
+		err := extractor.extractTypes(ctx, schemaPath, schema, result.resources, result.otherTypes)
+		if err != nil {
+			return swaggerTypes{}, errors.Wrapf(err, "error processing %q", schemaPath)
+		}
+	}
+
+	return result, nil
+}
+
+// loadAllSchemas walks all .json files in the given rootPath in directories
+// of the form "Microsoft.GroupName/…/2000-01-01/…" (excluding those matching
+// skipDirectories), and returns those files in a map of path→swagger spec.
+func loadAllSchemas(
+	ctx context.Context,
+	rootPath string) (map[string]spec.Swagger, error) {
+
+	var eg errgroup.Group
+
+	var mutex sync.Mutex
+	schemas := make(map[string]spec.Swagger)
+
+	err := filepath.Walk(rootPath, func(filePath string, fileInfo os.FileInfo, err error) error {
+
+		if err != nil {
+			return err
+		}
+
+		if ctx.Err() != nil {
+			return ctx.Err()
+		}
+
+		for _, skipDir := range skipDirectories {
+			if strings.Contains(filePath, skipDir) {
+				return filepath.SkipDir // this is a magic error
+			}
+		}
+
+		if !fileInfo.IsDir() &&
+			filepath.Ext(filePath) == ".json" &&
+			swaggerGroupRegex.MatchString(filePath) &&
+			swaggerVersionRegex.MatchString(filePath) {
+
+			// all files are loaded in parallel to speed this up
+			eg.Go(func() error {
+				var swagger spec.Swagger
+
+				fileContent, err := ioutil.ReadFile(filePath)
+				if err != nil {
+					return errors.Wrap(err, "unable to read swagger file")
```

Comment:
Do we need to include the path of the file in these errors?

Reply:
I was rather hopeful that the stdlib would already :)

Reply:
Done.